### PR TITLE
feat: fix compat with Rust, enable Ping + Identify

### PR DESCRIPTION
**Pull request by bajtos**
Enable handling of `Ping` and `Identify` requests by all Zinnia nodes.

This fixes compatibility with the rust-libp2p implementation of Ping protocol, which we need to allow us to test our request-response-like behavior against a Ping protocol implementation.

This is a follow-up for
- #83 
- #84

and a part of
- #73

### CODE REVIEW

File: ext/libp2p/peer.rs [hunk 1]
```diff
@@ -422,56 +468,62 @@ mod tests {
...
-            server_transport,
-            libp2p::ping::Behaviour::new(
-                libp2p::ping::Config::new()
-                    .with_max_failures(std::num::NonZeroU32::new(10).unwrap()),
-            ),
-            server_peer_id,
-        );
-        let server_task = {
+        let listener_addr: Multiaddr = "/ip4/127.0.0.1/tcp/10458".parse().unwrap();
+        listener_swarm.listen_on(listener_addr.clone()).unwrap();
```
Comment (bajtos) [inline ext/libp2p/peer.rs:493]: @mxinden What is the canonical way for setting up a Swarm listening on an ephemeral port?

I want to write something like this - notice the port is 0 (`/tcp/0`):

```rs
let listener_addr: Multiaddr = "/ip4/127.0.0.1/tcp/10458".parse().unwrap();
listener_swarm.listen_on(listener_addr.clone()).unwrap();
// 1. wait until OS binds the port a specific number
// 2. get the updated multi address
// E.g. the function implementing (1) can return a future of the new multiaddr
```

Comment (mxinden) [reply ext/libp2p/peer.rs:493]: You can call `listen_on("/ip4/127.0.0.1/tcp/0".parse().unwrap()).unwrap()` and then wait for a `SwarmEvent::NewListenAddr`.

File: ext/libp2p/peer.rs [hunk 2]
```diff
@@ -422,56 +468,62 @@ mod tests {
         init();
         let cancellation_token = CancellationToken::new();
 
-        let server_id_keys = identity::Keypair::generate_ed25519();
-        let server_peer_id = server_id_keys.public().to_peer_id();
-        let mut server_transport = create_transport(&server_id_keys).unwrap();
+        let listener_id_keys = identity::Keypair::generate_ed25519();
```
Comment (mxinden) [inline ext/libp2p/peer.rs:471]: Note that the new `libp2p-swarm-test` crate can help with all this boilerplate:

https://github.com/libp2p/rust-libp2p/tree/master/swarm-test

**Review by mxinden — COMMENTED** [review]
Sorry for the delayed review.

File: ext/libp2p/peer.rs (hunk 2 above)
Comment (bajtos) [reply ext/libp2p/peer.rs:471]: @mxinden thank you for the tip, the helpers in the swarm-test crate look really helpful! I would love to use [`listen`](https://github.com/libp2p/rust-libp2p/blob/32598d89d6a3008e416679ca4c2705d8b439c541/swarm-test/src/lib.rs#L77), see https://github.com/filecoin-station/zinnia/pull/85#discussion_r1123115694

There is a catch though: the swarm-test crate is not re-exported by libp2p and it's not published as a standalone crate to crates.io either :(

Comment (bajtos) [reply ext/libp2p/peer.rs:471]: Do you know how I can use `swarm-test` from my code?

**Pull request merged**